test/spec/unit/core/bidderFactory_spec.js

```diff
@@ -1022,6 +1023,166 @@ describe('validate bid response: ', function () {
+
+      expect(addBidResponseStub.calledOnce).to.equal(true);
+      expect(logWarnSpy.callCount).to.equal(0);
+      expect(logErrorSpy.callCount).to.equal(0);
+    });
+
+    it('should accept the bid, when unknowBidderCodes is empty and allowUknownBidderCode flag is true', function () {
+      bidderSettingStub.withArgs(CODE, 'allowUnknownBidderCodes').returns(true);
+      bidderSettingStub.withArgs(CODE, 'allowedUnknownBidderCodes').returns(['*']);
+
+      const bidder = newBidder(spec);
+      spec.interpretResponse.returns(bids1);
+      bidder.callBids(bidRequest, addBidResponseStub, doneStub, ajaxStub, onTimelyResponseStub, wrappedCallback);
+
+      expect(addBidResponseStub.calledOnce).to.equal(true);
+      expect(logWarnSpy.callCount).to.equal(0);
+      expect(logErrorSpy.callCount).to.equal(0);
+    });
+
+    it('should accept the bid, when unknowBidderCodes has bidder name and allowUknownBidderCode flag is true', function () {
+      bidderSettingStub.withArgs(CODE, 'allowUnknownBidderCodes').returns(true);
+      bidderSettingStub.withArgs(CODE, 'allowedUnknownBidderCodes').returns(['unknownBidder']);
+
+      const bidder = newBidder(spec);
+      spec.interpretResponse.returns(bids1);
+      bidder.callBids(bidRequest, addBidResponseStub, doneStub, ajaxStub, onTimelyResponseStub, wrappedCallback);
+
+      expect(addBidResponseStub.calledOnce).to.equal(true);
+      expect(logWarnSpy.callCount).to.equal(0);
+      expect(logErrorSpy.callCount).to.equal(0);
+    });
+
+    it('should accept the bid, when bidder is an alias of the adapter', function () {
+      bidderSettingStub.withArgs(CODE, 'allowUnknownBidderCodes').returns(false);
+      aliasRegistry = {'unknownBidder': 'knownAdapter1'};
+
+      const bidder = newBidder(spec);
+      spec.interpretResponse.returns(bids1);
+      bidder.callBids(bidRequest, addBidResponseStub, doneStub, ajaxStub, onTimelyResponseStub, wrappedCallback);
+
+      expect(addBidResponseStub.calledOnce).to.equal(true);
+      expect(logWarnSpy.callCount).to.equal(0);
+      expect(logErrorSpy.callCount).to.equal(0);
+    });
+
+    it('should not accept the bid, when bidder is an alias of the other adapter and allowUnknownBidderCodes is false', function () {
+      bidderSettingStub.withArgs(CODE, 'allowUnknownBidderCodes').returns(false);
+      aliasRegistry = {'unknownBidder': 'unknownAdapter1'};
+      adapterManagerStub.withArgs('unknownBidder').returns('unknownAdapter1')
+
+      const bidder = newBidder(spec);
+      spec.interpretResponse.returns(bids1);
+      bidder.callBids(bidRequest, addBidResponseStub, doneStub, ajaxStub, onTimelyResponseStub, wrappedCallback);
+
+      expect(addBidResponseStub.calledOnce).to.equal(false);
+      expect(logWarnSpy.callCount).to.equal(1);
+    });
+
+    it('should not accept the bid, when bidder is an alias of the other adapter and allowUnknownBidderCodes is true', function () {
```

Comment:
Same question - why? If I set `{allowUnknownBidderCodes: true, allowedUnknownBidderCodes: ['*']}` like this test does, shouldn't every bid always be accepted?

Reply:
I disagree - on two fronts: first, we already have bidders that want to bid on behalf of others, so I don't think it's that implausible that they may bid for an alias of them. Second, I don't think we need to contradict what the publisher has set; if they explicitly want to allow `appnexusAST`, we should allow it, I don't see a benefit in adding exceptions to their rules.